core/src/main/java/org/frankframework/util/MessageBrowsingUtil.java

```diff
@@ -34,27 +33,25 @@ public static String getMessageText(RawMessageWrapper<?> rawMessageWrapper, ILis
 			return null;
 		}
 
-		if(rawMessageWrapper instanceof MessageWrapper messageWrapper) {
+		if (rawMessageWrapper instanceof MessageWrapper messageWrapper) {
 			return messageWrapper.getMessage().asString();
 		}
 		Object rawMessage = rawMessageWrapper.getRawMessage();
-		if(rawMessage instanceof Message message) { // For backwards compatibility: earlier MessageLog messages were stored as Message.
+		if (rawMessage instanceof Message message) { // For backwards compatibility: earlier MessageLog messages were stored as Message.
 			return message.asString();
-		} else if(rawMessage instanceof String string) { // For backwards compatibility: earlier MessageLog messages were stored as String.
+		} else if (rawMessage instanceof String string) { // For backwards compatibility: earlier MessageLog messages were stored as String.
 			return string;
 		} else if (listener != null) {
-			String msg = null;
 			try {
-				msg = listener.extractMessage(rawMessageWrapper, new HashMap<>()).asString();
+				String msg = listener.extractMessage(rawMessageWrapper, new HashMap<>()).asString();
+				if (StringUtils.isNotEmpty(msg)) {
+					return msg;
+				}
 			} catch (Exception e) {
 				log.warn("{} cannot extract raw message [{}] ({}): {}", ClassUtils.nameOf(listener), rawMessageWrapper, ClassUtils.nameOf(e), e.getMessage(), e);
 			}
-			if (StringUtils.isEmpty(msg)) {
-				msg = Message.asString(rawMessage);
-			}
-			return msg;
-		} else {
-			return Message.asString(rawMessage);
 		}
+
+		return MessageUtils.asString(rawMessage);
```

Comment:
I wonder if this code is ever triggered, or what we would need to do in testing to trigger this.

Reply:
Same, but that's not for this pr 😄